server/fishtest/rundb.py

```diff
@@ -1163,6 +1175,46 @@ def count_games(d):
 
         return ret
 
+    def check_results(self, run, run_id, task_id):
+        old = run["results"]
+
+        # Force recalculation of results
+        run["results_stale"] = True
+
+        # Recalculate results from all tasks in run["tasks"].
+        # Sets run["results_stale"]=False and calls buffer(True).
+        self.get_results(run, True)
+
+        # Log any discrepancies between incremented and recalculated results
+        new = run["results"]
+        for s in ["wins", "losses", "draws", "crashes", "time_losses"]:
+            if old.get(s, -1) != new.get(s, -1):
+                info = "Check_results: task {}/{} {} results mismatch: {}/{}".format(
+                    run_id, task_id, s, old.get(s, -1), new.get(s, -1)
+                )
+                print(info, flush=True)
+
+        if (
+            "pentanomial" not in old
+            or "pentanomial" not in new
+            or len(old["pentanomial"]) < 5
+            or len(new["pentanomial"]) < 5
+        ):
+            info = "Check_results: task {}/{} pentanomial length results mismatch: {}/{}".format(
+                run_id,
+                task_id,
+                len(old.get("pentanomial", [])),
+                len(new.get("pentanomial", [])),
+            )
+            print(info, flush=True)
+        else:
```

Comment:
This should be equivalent.
```python
        else:
            for i, (old_value, new_value) in enumerate(
                zip(old["pentanomial"], new["pentanomial"])
            ):
                if old_value != new_value:
                    info = "Check_results: task {}/{} pentanomial value {} results mismatch: {}/{}".format(
                        run_id, task_id, i, old_value, new_value
                    )
                    print(info, flush=True)
```

Reply:
Yes, nice again.

I misread it when I looked earlier and thought I might disagree, but ofc you are right.

Thanks for your help! :-)